transport/internal/tls/dialer/dialer_test.go

```diff
@@ -91,14 +111,42 @@ func TestDialer(t *testing.T) {
 			_, err = conn.Read(buf)
 			require.NoError(t, err)
 			assert.Equal(t, tt.data, string(buf))
-
+			assertMetrics(t, root, false)
 			if tt.withCustomDialer {
 				assert.True(t, calledCustomDialer)
 			}
 		})
 	}
 }
 
+func assertMetrics(t *testing.T, root *metrics.Root, handshakeFailure bool) {
+	var expectedCounter metrics.Snapshot
+	expectedTags := metrics.Tags{
+		"service":   "test-svc",
+		"transport": "test-transport",
+		"component": "yarpc",
+		"mode":      "Enforced",
+		"direction": "outbound",
+		"dest":      "test-dest",
+	}
+	if handshakeFailure {
+		expectedCounter = metrics.Snapshot{
+			Name:  "tls_handshake_failures",
+			Tags:  expectedTags,
+			Value: 1,
+		}
+	} else {
+		expectedTags["version"] = "1.3"
+		expectedCounter = metrics.Snapshot{
+			Name:  "tls_connections",
+			Tags:  expectedTags,
+			Value: 1,
```

Comment:
can we move those 2 lines to L123 for common initialization, and just send `Name` here.